contribute/design/proposal-cstor-orchestration.md

```diff
@@ -0,0 +1,36 @@
+# Managing cStor based OpenEBS Volumes. 
+
+## Background 
+
+
+## Design
+
+Provisioning of cStor based Volume involve two steps. 
+
+1. Creating cStor StoragePools:
+
+- Admin will create SPC that contains information like:
+  - type=block
+  - disks list (type of PDC or specific disks)
+  - nodes list (optional)
+
+- maya-cstor-operator (embedded into maya-apiserver), will be watching for SPCs (type=block). If nodes were not specified in the PVC, the maya-cstor-operator will query the kube-ectd for the availability of the disks and determine the nodes. 
+
+- maya-cstor-operator will create a cStorPoolConfig object -  (with information required to initialize the zpool) 
+
+- maya-cstor-operator will create a Deployment YAML file  that contains the cStor container, the associated sidecars.  The cStor-Sidecar is passed the “id” of the cStorPoolConfig object 
+
+- once the cStor pool (container) is started, maya-cstor-operator will create the SP object. 
+
+
+2. Creating Volume using the (cStor) Storage Pools:
+
+- maya-apiserver will identify the storagepools to use based on the pool information specified in the PVC/StorageClass
```

Comment:
More explanation will be helpful on how information in PVC/storageclass will be used to identify storagepools.

Reply:
however, I can see that above question is out of scope of this document. But, ask is for better picture of the flow :)

Reply:
cool. Added one more step. And later we will include the actual YAMLs..